test/gtest/conv_igemm_dynamic_xdlops.cpp

```diff
@@ -0,0 +1,221 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <tuple>
+#include <miopen/miopen.h>
+#include <gtest/gtest.h>
+#include "../conv2d.hpp"
+#include "get_handle.hpp"
+
+using TestCase = std::tuple<std::vector<std::string>, std::string>;
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_ALL)
+
+MIOPEN_DECLARE_ENV_VAR(MIOPEN_TEST_GPU_XNACK_ENABLED)
+
+static bool SkipTest(void)
+{
+    return miopen::IsEnabled(MIOPEN_TEST_GPU_XNACK_ENABLED{}) ||
+           miopen::IsDisabled(MIOPEN_TEST_ALL{});
+}
+
+void GetArgs(const TestCase& param, std::vector<std::string>& tokens)
+{
+    auto env_vars = std::get<0>(param);
+    for(auto& elem : env_vars)
+    {
+        putenv(elem.data());
+    }
+
+    auto cmd = std::get<1>(param);
+
+    std::stringstream ss(cmd);
+    std::istream_iterator<std::string> begin(ss);
+    std::istream_iterator<std::string> end;
+    while(begin != end)
+        tokens.push_back(*begin++);
+}
+
+class Conv2dFloat : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+class Conv2dHalf : public testing::TestWithParam<std::vector<TestCase>>
+{
+};
+
+void Run2dDriver(miopenDataType_t prec)
+{
+
+    std::vector<TestCase> params;
+    switch(prec)
+    {
+    case miopenFloat: params = Conv2dFloat::GetParam(); break;
+    case miopenHalf: params = Conv2dHalf::GetParam(); break;
+    case miopenInt8:
+    case miopenBFloat16:
+    case miopenInt32:
+    case miopenDouble:
+    case miopenFloat8:
+    case miopenBFloat8:
+        FAIL() << "miopenInt8, miopenBFloat16, miopenInt32, "
+                  "miopenDouble, miopenFloat8, miopenBFloat8 "
+                  "data type not supported by conv_igemm_dynamic_xdlops test";
+
+    default: params = Conv2dFloat::GetParam();
+    }
+
+    for(const auto& test_value : params)
+    {
+        std::vector<std::string> tokens;
+        GetArgs(test_value, tokens);
+        std::vector<const char*> ptrs;
+
+        std::transform(tokens.begin(),
+                       tokens.end(),
+                       std::back_inserter(ptrs),
+                       [](const std::string& str) { return str.data(); });
+
+        testing::internal::CaptureStderr();
+        test_drive<conv2d_driver>(ptrs.size(), ptrs.data());
```

Comment:
test_drive expects the first char* in the argv to be the program name, but it looks like there is no program name in the test cases, or inserted at any later point. This may dump the first argument.